packages/common/src/extensions/slickRowBasedEdit.ts

```diff
@@ -229,6 +239,45 @@ export class SlickRowBasedEdit {
     });
   }
 
+  translate() {
+    this.btnUpdateTitle =
+      (this.gridOptions.rowBasedEditOptions?.actionButtons?.updateButtonTitleKey &&
+        this.extensionUtility.translaterService?.translate?.(
+          this.gridOptions.rowBasedEditOptions?.actionButtons?.updateButtonTitleKey
+        )) ||
+      this.gridOptions.rowBasedEditOptions?.actionButtons?.updateButtonTitle ||
+      'Update the Row';
```

Comment:
I see a lot of the same code repeated, perhaps we could migrate this to an external protected function? Something along these lines

```ts
protected getTitleOrDefault(key: string, defaultTitle: string) {
  const actionBtnOptions = this.gridOptions.rowBasedEditOptions?.actionButtons; 
  const hasTranslateService = this.extensionUtility.translaterService?.translate;

  // another approach might be to move the line above to GETTERs in the class?
  
  return hasTranslateService ? actionBtnOptions[key] : defaultTitle;
}

this.btnUpdateTitle = this.getTitleOrDefault('updateButtonTitleKey', 'Update the Row');
```

I should probably do that in some other extensions like the GridMenu that I think I did something like you did and I think that this could make the code more readable.